### PR TITLE
NOISSUE - Make Check Interval Configurable

### PR DESCRIPTION
**Pull request by rodneyosodo**
# What type of PR is this?

This is a bug fix because it fixes failing tests for events package

## What does this do?

This PR makes check interval for unpublished events configurable for redis since Redis is the only event source than that doesn't support unavailable publish i.e store events to a buffer if the server is down

## Which issue(s) does this PR fix/relate to?

No issue

## Have you included tests for your changes?

Yes, I have included tests for my changes.

## Did you document any new/modified feature?

No

### CODE REVIEW

**Review by dborovcanin — CHANGES_REQUESTED** [review]
Did you try to spawn fewer routines (say, 100, instead of 1m), and wait shorter (say, 3s instead of 1m)? Also, consider replacing simple wait with the actual consumer (while this will definitely make this test even more "integration") so that we wait for all the events to be consumed properly.

File: pkg/events/redis/publisher.go [hunk 1]
```diff
@@ -16,22 +16,24 @@ import (
...
-	mu                sync.Mutex
+	client                         *redis.Client
+	unpublishedEvents              chan *redis.XAddArgs
+	stream                         string
+	mu                             sync.Mutex
+	unpublishedEventsCheckInterval time.Duration
 }
 
-func NewPublisher(ctx context.Context, url, stream string) (events.Publisher, error) {
+func NewPublisher(ctx context.Context, url, stream string, unpublishedEventsCheckInterval time.Duration) (events.Publisher, error) {
```
Comment (dborovcanin) [inline pkg/events/redis/publisher.go:26]: Rename to `retryPeriod` or just `retry` - `time.Duration` type implies the purpose.

File: pkg/events/redis/publisher.go [hunk 2]
```diff
@@ -71,16 +73,18 @@ func (es *pubEventStore) Publish(ctx context.Context, event events.Event) error
 	}
 }
 
+// startPublishingRoutine periodically checks the Redis connection and publishes
```
Comment (dborovcanin) [inline pkg/events/redis/publisher.go:76]: Rename to `flushUnpublished` or something similar.

File: pkg/events/redis/publisher.go [hunk 3]
```diff
@@ -71,16 +73,18 @@ func (es *pubEventStore) Publish(ctx context.Context, event events.Event) error
...
 
-	ticker := time.NewTicker(events.UnpublishedEventsCheckInterval)
+	ticker := time.NewTicker(es.unpublishedEventsCheckInterval)
 	defer ticker.Stop()
 
 	for {
 		select {
 		case <-ticker.C:
-			if err := es.checkRedisConnection(ctx); err == nil {
+			if err := es.checkConnection(ctx); err == nil {
```
Comment (dborovcanin) [inline pkg/events/redis/publisher.go:87]: Any reason for reversing the order here?

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: pkg/events/redis/publisher.go (hunk 3 above)
Comment (rodneyosodo) [reply pkg/events/redis/publisher.go:87]: The order hasn't changed only renaming

Comment (dborovcanin) [reply pkg/events/redis/publisher.go:87]: I meant, [on the line below](https://github.com/absmach/magistrala-old/blob/867060f6a5c76d4cf8f8dbe632da76283896cd63/pkg/events/redis/publisher.go#L85); we use reverse order to publish events. Is there a reason we use reverse order?

**Review by dborovcanin — APPROVED** [review]

File: pkg/events/redis/publisher.go (hunk 3 above)
Comment (rodneyosodo) [reply pkg/events/redis/publisher.go:87]: First event published will be the first event to be republished

**Pull request merged**